airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/catalog_assets.py

```diff
@@ -0,0 +1,161 @@
+
+def render_connector_catalog_locations_html(destinations_table, sources_table):
+    env = Environment(loader=PackageLoader("orchestrator", "templates"))
+    template = env.get_template("connector_catalog_locations.html")
+    return template.render(destinations_table=destinations_table, sources_table=sources_table)
+
+def render_connector_catalog_locations_markdown(destinations_markdown, sources_markdown):
+    env = Environment(loader=PackageLoader("orchestrator", "templates"))
+    template = env.get_template("connector_catalog_locations.md")
+    return template.render(destinations_markdown=destinations_markdown, sources_markdown=sources_markdown)
+
+@asset(required_resource_keys={"catalog_report_directory_manager"})
+def connector_catalog_location_html(context, all_destinations_dataframe, all_sources_dataframe):
+    """
+    Generate an HTML report of the connector catalog locations.
+    """
+
+    columns_to_show = ["name_x", "dockerRepository", "dockerImageTag", "is_oss", "is_cloud"]
+
+    # convert true and false to checkmarks and x's
+    all_sources_dataframe.replace({True: "✅", False: "❌"}, inplace=True)
+    all_destinations_dataframe.replace({True: "✅", False: "❌"}, inplace=True)
+
+    html = render_connector_catalog_locations_html(
+        destinations_table=all_destinations_dataframe[columns_to_show].to_html(),
+        sources_table=all_sources_dataframe[columns_to_show].to_html(),
+    )
+
+    catalog_report_directory_manager = context.resources.catalog_report_directory_manager
+    file_handle = catalog_report_directory_manager.write_data(html.encode(), ext="html", key="connector_catalog_locations")
+
+    metadata = {
+        "preview": html,
+        "gcs_path": MetadataValue.url(file_handle.gcs_path),
+    }
+
+    return Output(metadata=metadata, value=file_handle)
+
+
+@asset(required_resource_keys={"catalog_report_directory_manager"})
+def connector_catalog_location_markdown(context, all_destinations_dataframe, all_sources_dataframe):
+    """
+    Generate a markdown report of the connector catalog locations.
+    """
+
+    columns_to_show = ["name_x", "dockerRepository", "dockerImageTag", "is_oss", "is_cloud"]
+
+    # convert true and false to checkmarks and x's
+    all_sources_dataframe.replace({True: "✅", False: "❌"}, inplace=True)
+    all_destinations_dataframe.replace({True: "✅", False: "❌"}, inplace=True)
+
+    markdown = render_connector_catalog_locations_markdown(
+        destinations_markdown=all_destinations_dataframe[columns_to_show].to_markdown(),
+        sources_markdown=all_sources_dataframe[columns_to_show].to_markdown(),
+    )
+
+    catalog_report_directory_manager = context.resources.catalog_report_directory_manager
+    file_handle = catalog_report_directory_manager.write_data(markdown.encode(), ext="md", key="connector_catalog_locations")
+
+    metadata = {
+        "preview": MetadataValue.md(markdown),
+        "gcs_path": MetadataValue.url(file_handle.gcs_path),
```

Comment:
How does this know to upload to GCS? Is this just something dagster has support for internally via this `gcs_path` attribute?

Reply:
All part of the resource we pull in "catalog_report_directory_manager"

You can see the gcs stuff all get wired up in the resources folder